File: pedidos-service/src/models/producto.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany, CreateDateColumn } from "typeorm";
import { Restaurante } from "./restaurante";
import { Reserva } from "./reserva";

@Entity()
export class Producto {
  @PrimaryGeneratedColumn()
  id_producto!: number;

  @Column()
  nombre!: string;

  @Column("text", { nullable: true })
  descripcion?: string;

  @Column("decimal", { precision: 10, scale: 2 })
  precio_original!: number;

  @Column("decimal", { precision: 10, scale: 2 })
  precio_descuento!: number;

  @Column()
  cantidad_disponible!: number;

  @Column({ type: "timestamp" })
  fecha_caducidad!: Date;

  @Column({ default: true })
  activo!: boolean;

  @Column({ nullable: true })
  imagen_url?: string;

  @CreateDateColumn()
  fecha_creacion!: Date;

  @ManyToOne(() => Restaurante, restaurante => restaurante.productos)
  restaurante!: Restaurante;

  @OneToMany(() => Reserva, reserva => reserva.producto)
  reservas!: Reserva[];
}

File: ofertas_services/models/producto.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Boolean, Text, ForeignKey
from sqlalchemy.orm import relationship
from database import Base

class Producto(Base):
    __tablename__ = "producto"

    id_producto = Column(Integer, primary_key=True, index=True)  # igual que TS
    nombre = Column(String, nullable=False)
    descripcion = Column(Text, nullable=True)
    precio_original = Column(Float, nullable=False)
    precio_descuento = Column(Float, nullable=False)
    cantidad_disponible = Column(Integer, nullable=False)
    fecha_caducidad = Column(DateTime, nullable=False)
    activo = Column(Boolean, default=True)
    imagen_url = Column(String, nullable=True)
    fecha_creacion = Column(DateTime)

    # Relación con restaurante (opcional si tienen modelo Restaurante en Python)
    restaurante_id = Column(Integer, ForeignKey("restaurante.id"))  # asumiendo
    # restaurante = relationship("Restaurante", back_populates="productos")

    # Reservas (si tienes modelo Reserva)
    # reservas = relationship("Reserva", back_populates="producto")
